msprime/demography.py

```diff
@@ -1754,6 +1813,82 @@ def from_old_style(
                 population_map,
             )
 
+    @staticmethod
+    def from_demes(graph):
+        def get_growth_rate(epoch):
+            ret = 0
+            if epoch.size_function not in ["constant", "exponential"]:
+                raise ValueError(
+                    "msprime only supports constant exponentially changing "
+                    "population sizes"
+                )
+            if epoch.end_size != epoch.start_size:
+                ret = -math.log(epoch.start_size / epoch.end_size) / epoch.time_span
+            return ret
+
+        graph = graph.in_generations()
+
+        demography = Demography()
+        for deme in graph.demes:
+            initial_size = 0
+            growth_rate = 0
+            last_epoch = deme.epochs[-1]
+            if last_epoch.end_time == 0:
+                initial_size = last_epoch.end_size
+                growth_rate = get_growth_rate(last_epoch)
+            demography.add_population(
+                name=deme.name,
+                description=deme.description,
+                growth_rate=growth_rate,
+                initial_size=initial_size,
+                default_sampling_time=deme.end_time,
+            )
+            for epoch in deme.epochs:
```

Comment:
Shouldn't this be `for epoch in reversed(deme.epochs[:-1]):`? The ZigZag model is a good many-epochs test case.

Reply:
I don't think it matters - we're sorting the events at the end anyway.

Reply:
Ah, I see what you mean now. Hmm, not sure how the zigzag models works at the moment then.